internal/charmstore/search.go

```diff
@@ -483,6 +493,23 @@ func (sp *SearchParams) ParseSortFields(f ...string) error {
 	return nil
 }
 
+func (sp *ListParams) ParseSortFieldsList(f ...string) error {
+	for _, s := range f {
+		for _, s := range strings.Split(s, ",") {
+			sort := sortListFields[s]
```

Comment:
You probably want to do something like this:

    field := sortListFields[strings.TrimPrefix(s, "-")]
    if field == "" {
         return errgo.Newf("unknown field %q", field)
    }
    if s[0] == '-' {
        field = "-" + field
    }
    sp.sort = append(sp.sort, field)